app/controllers/nonsipp/common/CompanyRecipientCrnController.scala

```diff
@@ -78,9 +80,12 @@ class CompanyRecipientCrnController @Inject()(
               .successful(BadRequest(view(formWithErrors, viewModel(srn, index, mode, subject, request.userAnswers)))),
           value =>
             for {
-              updatedAnswers <- Future
-                .fromTry(request.userAnswers.set(CompanyRecipientCrnPage(srn, index, subject), ConditionalYesNo(value)))
-              _ <- saveService.save(updatedAnswers)
+              updatedAnswers <- request.userAnswers
+                .set(CompanyRecipientCrnPage(srn, index, subject), ConditionalYesNo(value))
+                .mapK
+              nextPage = navigator.nextPage(CompanyRecipientCrnPage(srn, index, subject), mode, updatedAnswers)
+              updatedProgressAnswers <- saveProgress(srn, index, updatedAnswers, nextPage)
+              _ <- saveService.save(updatedProgressAnswers)
             } yield Redirect(navigator.nextPage(CompanyRecipientCrnPage(srn, index, subject), mode, updatedAnswers))
```

Comment:
This can use `nextPage` defined above.

Reply:
This has been addressed in a code cleanup , commit needs to yet be pushed